 caching/metadata.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/caching/metadata.cc b/caching/metadata.cc
index 7d8bfae..5457533 100644
--- a/caching/metadata.cc
+++ b/caching/metadata.cc
@@ -124,6 +124,10 @@ void
 metadata::commit_mappings()
 {
 	sb_.mapping_root = mappings_->get_root();
+	if (sb_.version >= 2) {
+		dirty_bits_->flush();
+		sb_.dirty_root = dirty_bits_->get_root();
+	}
 }
 
 void
