--- thin-provisioning-tools-0.2.8/Makefile.in.orig	2014-04-14 11:42:09.530804553 +0200
+++ thin-provisioning-tools-0.2.8/Makefile.in	2014-04-14 11:42:42.455259135 +0200
@@ -20,11 +20,6 @@
 
 V=@
 PROGRAMS=\
-	cache_check \
-	cache_dump \
-	cache_restore \
-	cache_repair \
-	\
 	thin_check \
 	thin_dump \
 	thin_restore \
@@ -77,9 +72,6 @@
 PDATA_OBJECTS=$(subst .cc,.o,$(SOURCE))
 
 CXX_PROGRAM_SOURCE=\
-	caching/cache_check.cc \
-	caching/cache_restore.cc \
-	\
 	thin-provisioning/thin_check.cc \
 	thin-provisioning/thin_dump.cc \
 	thin-provisioning/thin_restore.cc \
@@ -264,22 +256,6 @@
 CACHE_RESTORE_SOURCE=$(SOURCE)
 CACHE_RESTORE_OBJECTS=$(subst .cc,.o,$(CACHE_RESTORE_SOURCE))
 
-cache_check: $(CACHE_CHECK_OBJECTS) caching/cache_check.o
-	@echo "    [LD]  $@"
-	$(V) $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@ $+ $(LIBS) $(LIBEXPAT)
-
-cache_dump: $(CACHE_DUMP_OBJECTS) caching/cache_dump.o
-	@echo "    [LD]  $@"
-	$(V) $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@ $+ $(LIBS) $(LIBEXPAT)
-
-cache_repair: $(CACHE_REPAIR_OBJECTS) caching/cache_repair.o
-	@echo "    [LD]  $@"
-	$(V) $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@ $+ $(LIBS) $(LIBEXPAT)
-
-cache_restore: $(CACHE_RESTORE_OBJECTS) caching/cache_restore.o
-	@echo "    [LD]  $@"
-	$(V) $(CXX) $(CXXFLAGS) $(LDFLAGS) -o $@ $+ $(LIBS) $(LIBEXPAT)
-
 DEPEND_FILES=\
 	$(subst .cc,.d,$(SOURCE)) \
 	$(subst .cc,.d,$(TEST_SOURCE)) \
@@ -299,10 +275,6 @@
 
 install: $(PROGRAMS)
 	$(INSTALL_DIR) $(BINDIR)
-	$(INSTALL_PROGRAM) cache_check $(BINDIR)
-	$(INSTALL_PROGRAM) cache_dump $(BINDIR)
-	$(INSTALL_PROGRAM) cache_repair $(BINDIR)
-	$(INSTALL_PROGRAM) cache_restore $(BINDIR)
 	$(INSTALL_PROGRAM) thin_check $(BINDIR)
 	$(INSTALL_PROGRAM) thin_dump $(BINDIR)
 	$(INSTALL_PROGRAM) thin_repair $(BINDIR)
@@ -310,10 +282,6 @@
 	$(INSTALL_PROGRAM) thin_rmap $(BINDIR)
 	$(INSTALL_PROGRAM) thin_metadata_size $(BINDIR)
 	$(INSTALL_DIR) $(MANPATH)/man8
-	$(INSTALL_DATA) man8/cache_check.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/cache_dump.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/cache_repair.8 $(MANPATH)/man8
-	$(INSTALL_DATA) man8/cache_restore.8 $(MANPATH)/man8
 	$(INSTALL_DATA) man8/thin_check.8 $(MANPATH)/man8
 	$(INSTALL_DATA) man8/thin_dump.8 $(MANPATH)/man8
 	$(INSTALL_DATA) man8/thin_repair.8 $(MANPATH)/man8
