--- thin-provisioning-tools-0.3.2/persistent-data/data-structures/bloom_filter.h.orig	2014-04-04 15:28:05.138763284 +0200
+++ thin-provisioning-tools-0.3.2/persistent-data/data-structures/bloom_filter.h	2014-04-04 15:28:09.715822655 +0200
@@ -12,7 +12,7 @@
 	class bloom_filter {
 	public:
 		typedef boost::shared_ptr<bloom_filter> ptr;
-		typedef typename persistent_data::transaction_manager::ptr tm_ptr;
+		typedef persistent_data::transaction_manager::ptr tm_ptr;
 
 		// nr_bits must be a power of two
 		bloom_filter(tm_ptr tm,
